Docs: continental extension cookbook

naliboff opened the pull request:
This PR updates the continental extension cookbook documentation, and supersedes #4734 which was initialized from the wrong repository by accident. 

* [x] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

naliboff commented:
@cmills1095 - Can you take a look when if you have a chance sometime in the next few days?

cmills1095 commented on cookbooks/continental_extension/doc/continental_extension.md:
````diff
@@ -1,39 +1,39 @@
+         modeling complex lithospheric processes. Such a material model can be used in
+       using the following set of input parameters:
+
+         ``` {literalinclude} continental_extension_material_model.prm
````
There's something weird going on with the indentation that's causing a lot of these directives to not build properly. It looks like you intentionally added these indents in the latest commit, can I ask why?

naliboff replied:
@cmills1095 - sorry about that. `make indent` was altering additional files aside from this one, so I tried running style on just this file. That clearly did not work, and the update commit hopefully has the correct indentation.

cmills1095 commented:
Not sure what happened to the indentation but I think everything will look a lot better if you undo the latest commit

cmills1095 commented on cookbooks/continental_extension/doc/continental_extension.md:
```diff
@@ -29,11 +29,10 @@ The 2D Cartesian model spans 200 ($x$) by 100 ($y$) km and has a finite
-Unlike the crustal deformation cookbook (see
-Section&nbsp;{ref}`sec:cookbooks-crustal-deformation`, the mesh is not
-refined with time.
+Unlike the crustal deformation cookbook (see {ref}`sec:cookbooks-crustal-deformation`),
```
{ref}`sec:cookbooks:crustal-deformation` is how that section header is currently formatted, might have been recently changed

cmills1095 commented on cookbooks/continental_extension/doc/continental_extension.md:
````diff
@@ -67,48 +66,53 @@ cohesion through time based on user-specified bounds and magnitudes. The
 equations defining the initial geotherm (Chapman 1986) follow the form
+```{math}
+\begin{aligned}
 $$\begin{aligned}
````
$$\begin{aligned} is redundant and causes an error

cmills1095 reviewed (changes requested):
Solid except the two comments (lines 32,77) and the citations, which should be formatted differently. Replace the references in these lines as such:
Lines 23 - 24: 
```
({cite}`Hui11,Bru14,Nal15` and references therein)
```
Line 59: 
```
{cite}`RB04`
```
Line 60: 
```
{cite}`RGWD06`
```
Line 60: 
```
{cite}`HK04`
```
Line 75: 
```
{cite}`Cha86`
```
Line 120 - 121: 
```
{cite}`Bru14`
```
Then delete all the reference text below line 137

naliboff commented:
@cmills1095 - Thanks for the review and suggestions! I've hopefully addressed all of the comments, but please let me know if further cleanup is needed.
